src/hooks/useSyncChainQuery.ts

```diff
@@ -21,8 +21,18 @@ function getParsedChainId(parsedQs?: ParsedQs) {
 }
 
 export default function useSyncChainQuery() {
-  const { chainId, isActive } = useWeb3React()
+  const { chainId, isActive, account } = useWeb3React()
   const parsedQs = useParsedQueryString()
+  const chainIdRef = useRef(chainId)
+  const accountRef = useRef(account)
+
+  useMemo(() => {
+    // Update chainIdRef when the account is retrieved from Web3React
+    if (account && account !== accountRef.current) {
+      chainIdRef.current = chainId
+      accountRef.current = account
+    }
+  }, [account, chainId])
```

Comment:
`useMemo` runs in a different stage than `useEffect`, and should not be used for side effects. Can this be outside of a `useMemo`?

Reply:
updated to useeffect